samples/template-operator/operator/controllers/sample_controller.go

```diff
@@ -85,20 +109,118 @@ func (r *SampleReconciler) HandleInitialState(ctx context.Context, sampleResourc
 	return r.Client.Status().Update(ctx, sampleResource)
 }
 
-func (r *SampleReconciler) HandleProcessingState(ctx context.Context, sampleResource *v1alpha1.Sample) error {
+func (r *SampleReconciler) HandleProcessingState(ctx context.Context, sampleResource *v1alpha1.Sample, logger *logr.Logger) error {
 	// TODO: processing logic here
 
-	// Example: If Processing succeeds, set to Ready state
-	sampleResource.Status.State = v1alpha1.SampleStateReady
-	return r.Client.Status().Update(ctx, sampleResource)
+	sampleObjUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(sampleResource)
+	if err != nil {
+		return err
+	}
+
+	manifestClient, err := r.getManifestClient(logger, sampleResource.Spec.ChartFlags)
+	if err != nil {
+		sampleResource.Status.State = v1alpha1.SampleStateError
+		return r.Client.Status().Update(ctx, sampleResource)
+	}
+
+	// Use manifest library client to install a sample chart
+	ready, err := manifestClient.Install(manifestLib.InstallInfo{
+		Ctx: ctx,
+		ChartInfo: &manifestLib.ChartInfo{
+			ChartPath:   sampleResource.Spec.ChartPath,
+			ReleaseName: sampleResource.Spec.ReleaseName,
+		},
+		RemoteInfo: custom.RemoteInfo{
```

Comment:
I think we should prepare the sample to run as a runtime / skr operator not a control plane operator, in this case, we wouldnt need this right?

Reply:
For now this is required to know the destination client, the name is a bis misleading I agree. I will refactor this in the manifest-library improve interface part.